pkg/controller/config/config_controller.go

```diff
@@ -185,17 +184,11 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 		if errors.IsNotFound(err) {
 			exists = false
 
-			// removing all items from the sync cache
-			// before sending the metric
+			// removing all items from the sync cache before sending the metric
 			for k := range r.syncMetricsCache.Cache {
-				delete(r.syncMetricsCache.Cache, k)
+				r.syncMetricsCache.DeleteObject(k)
```

Comment:
Why not just wipe the map and create a new one?

then you could use the name  `ResetCache()`, and it would use the same behavior we rely on below (removing all objects).

Reply:
thanks, added `ResetCache` to wipe the cache